ci/Dockerfile-envoy

```diff
@@ -2,12 +2,19 @@ FROM ubuntu:18.04
 
 RUN apt-get update \
     && apt-get upgrade -y \
-    && apt-get install -y ca-certificates \
-    && apt-get autoremove -y \
+    && apt-get install --no-install-recommends -y ca-certificates curl gcc libc-dev \
```

Comment:
I'd prefer not building inside docker image.

Reply:
me too. In terms of the `envoy` build pipeline im not sure where else to do it.

If you can suggest a better way, i can happily update.

Reply:
at least use multi-stage docker build and verify the source code sha256sum.

Reply:
i have updated the PR to do this, and pinned the fetch of `su-exec` source to current master